src/openms/source/CHEMISTRY/Residue.cpp

```diff
@@ -506,8 +506,8 @@ namespace OpenMS
   void Residue::setModification(const String& name)
   {
     ModificationsDB* mod_db = ModificationsDB::getInstance();
-    const ResidueModification& mod = mod_db->getModification(name, one_letter_code_, ResidueModification::ANYWHERE);
-    setModification_(mod);
+    const ResidueModification* mod = mod_db->getModification(name, one_letter_code_, ResidueModification::ANYWHERE);
+    setModification_(*mod);
```

Comment:
what if the mod does not exist?

Reply:
we could fatally exit the program with an error message 
or return an error code (leaving the mod untouched)